extensions/barrage/src/main/java/io/deephaven/extensions/barrage/chunk/vector/ByteVectorExpansionKernel.java

```diff
@@ -37,24 +39,31 @@ public <A extends Any> WritableChunk<A> expand(
         }
 
         final ObjectChunk<ByteVector, A> typedSource = source.asObjectChunk();
-        final SizedByteChunk<A> resultWrapper = new SizedByteChunk<>();
 
-        int lenWritten = 0;
+        long totalSize = 0;
+        for (int i = 0; i < typedSource.size(); ++i) {
+            final ByteVector row = typedSource.get(i);
+            totalSize += row == null ? 0 : row.size();
+        }
+        final WritableByteChunk<A> result = WritableByteChunk.makeWritableChunk(
+                LongSizedDataStructure.intSize("ExpansionKernel", totalSize));
+        result.setSize(0);
+
         perElementLengthDest.setSize(source.size() + 1);
         for (int i = 0; i < typedSource.size(); ++i) {
             final ByteVector row = typedSource.get(i);
-            final int len = row == null ? 0 : row.intSize("ByteVectorExpansionKernel");
-            perElementLengthDest.set(i, lenWritten);
-            final WritableByteChunk<A> result = resultWrapper.ensureCapacityPreserve(lenWritten + len);
-            for (int j = 0; j < len; ++j) {
-                result.set(lenWritten + j, row.get(j));
+            perElementLengthDest.set(i, result.size());
+            if (row == null) {
+                continue;
+            }
+            final ByteConsumer consumer = result::add;
```

Comment:
Charles made a claim recently that using set and a single setSize was more performant. I don't fully believe it, especially in a case like this one where the counter has to live somewhere.